### ISSUE TITLE
Unable to connect to Iot Hub from within an Azure Function app

### ISSUE THREAD

**Issue by kc-95-scg** [Azure/azure-iot-sdk-node#992]
# Context

- **OS and version used:** Local Machine (Ubuntu 20.04), Azure Function App (Windows, Node.js 14 LTS)
- **Node.js version:** Node.js 14 LTS
- **npm version:** 1.14.2
- **list of installed packages:** azure-iothub@1.14.2 only
- **cloned repo:** <VERSION> 

## Description of the issue

I am planning to create a HTTP trigger function as an action from an Iot Hub alert.

I am able to perform IotHub registry operations by running a normal .js script on a local machine connected to the internet, but not able to perform registry operations from within an Azure Function.

I have enabled public network access on my Iot Hub.

## Code sample exhibiting the issue

These are the only few lines of code that i am running. The exact same code is running on an azure function.
```
var iothub = require('azure-iothub');
var conn_str="HostName=<hostname>.azure-devices.net;SharedAccessKeyName=iothubowner;SharedAccessKey=<key>"
var registry = iothub.Registry.fromConnectionString(conn_str);
console.log(registry)

console.log("Before registry list")
registry.list(function (err, deviceList) {
    console.log(deviceList)
})
console.log("After registry list")
```
Further registry functions were not possible as well.

This is the only dependency in my package.json
```
  "dependencies": {
    "azure-iothub": "^1.14.2"
  }
```

## Console log of the issue

From my test_registry.js ( $ node test_registry.js)
```
Registry {
  _restApiClient:
   RestApiClient {
     _iotHubPublicScope: [ 'https://iothubs.azure.net/.default' ],
     _BearerTokenPrefix: 'Bearer ',
     _MinutesBeforeProactiveRenewal: 9,
     _MillisecsBeforeProactiveRenewal: 540000,
     _config:
      { host: '<host_name>.azure-devices.net',
        sharedAccessSignature: [SharedAccessSignature],
        tokenCredential: undefined },
     _userAgent: 'azure-iothub/1.14.2',
     _http: Http { _options: [Object] } } }
Before registry list
After registry list
[ Device {
..... .. Shows list of devices and properties on the iothub here. } ]
```

From Azure Function (same logs when running function locally and when deployed on the function app_
```
[2021-06-16T17:00:33.262Z] Registry {
[2021-06-16T17:00:33.262Z]   _restApiClient:
[2021-06-16T17:00:33.262Z]    RestApiClient {
[2021-06-16T17:00:33.262Z]      _iotHubPublicScope: [ 'https://iothubs.azure.net/.default' ],
[2021-06-16T17:00:33.262Z]      _BearerTokenPrefix: 'Bearer ',
[2021-06-16T17:00:33.262Z]      _MinutesBeforeProactiveRenewal: 9,
[2021-06-16T17:00:33.262Z]      _MillisecsBeforeProactiveRenewal: 540000,
[2021-06-16T17:00:33.263Z]      _config:
[2021-06-16T17:00:33.263Z]       { host: '<host_name>.azure-devices.net',
[2021-06-16T17:00:33.263Z]         sharedAccessSignature: [SharedAccessSignature],
[2021-06-16T17:00:33.263Z]         tokenCredential: undefined },
[2021-06-16T17:00:33.263Z]      _userAgent: 'azure-iothub/1.14.2',
[2021-06-16T17:00:33.263Z]      _http: Http { _options: [Object] } } }
[2021-06-16T17:00:33.264Z] Before registry list
[2021-06-16T17:00:33.264Z] After registry list
[2021-06-16T17:00:33.360Z] Executed 'Functions.<functionname>' (Succeeded, Id=<functionid> Duration=1191ms)
```

**Comment by anthonyvercolano** [comment]
Sorry for the delay.

What exactly do you mean "not able to perform registry operations from within an Azure Function"?

Are you able to generate logs inside the AF?
What errors happen in the AF?

**Comment by kc-95-scg** [comment]
@anthonyvercolano Thanks for the reply. I have raised a ticket on Azure for support and they have followed up with me on the issue.
Basically the .list() is an async function, and context.log cannot trace the function output.
I have managed to debug the code locally using console.logs then deploy it on the cloud and use it without any logs/printouts.

Turns out it was running fine all along, just the logs (or empty logs in this case) were misleading.
